python/ray/autoscaler/_private/util.py

```diff
@@ -97,7 +97,14 @@ def validate_config(config: Dict[str, Any]) -> None:
                 "sum of `min_workers` of all the available node types.")
 
 
-def prepare_config(config):
+def prepare_config(config: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    The returned config has the following properties:
+    - Uses the multi-node-type autoscaler configuration.
+    - Merged with the appropriate defaults.yaml
+    - Has a valid Docker configuration if provided.
+    - Has max_worker set for each node type.
+    """
```

Comment:
@DmitriGekhtman AFAIK, after `fillout_defaults` is called, `available_node_types` must be in use because either:
1. The user provided them
2. The user excluded them (and they get filled in from the `defaults.yaml`)
3. The user provided `available_node_types = null`, but this will fail [validation](https://github.com/ray-project/ray/blob/34b5781ae0d46b769e8bac400e0cbad6c60c8fcd/python/ray/autoscaler/_private/util.py#L87) and should be considered a pathological case.

Reply:
That's right.